src/main/java/org/skriptlang/skript/lang/script/Script.java

```diff
@@ -219,4 +222,38 @@ public <T extends ScriptEvent> Set<T> getEvents(Class<T> type) {
 		);
 	}
 
+	@Override
+	public boolean hasExperiment(Experiment experiment) {
+		@Nullable ExperimentSet set = this.getData(ExperimentSet.class);
+		if (set == null)
+			return false;
+		return set.contains(experiment);
+	}
+
+	/**
+	 * Marks this as using an experimental feature.
+	 * @param experiment The feature to register.
+	 */
+	@ApiStatus.Internal
+	public void addExperiment(Experiment experiment) {
```

Comment:
I wonder if we should restrict modifying the experiments a script is using to parse time. It seems like that is the expectation (for example, CondIsUsingFeature calculates the value if possible during init)

To do this, you can check `ParserInstance#isActive`